zipline/finance/execution.py

```diff
@@ -158,19 +189,31 @@ def asymmetric_round_price_to_penny(price, prefer_round_down,
     If prefer_round_down: [<X-1>.0095, X.0195) -> round to X.01.
     If not prefer_round_down: (<X-1>.0005, X.0105] -> round to X.01.
     """
+    precision = get_precision(tick_size)
+    diff *= (10 ** -precision)
+    print(price, precision, diff)
+
     # Subtracting an epsilon from diff to enforce the open-ness of the upper
     # bound on buys and the lower bound on sells.  Using the actual system
     # epsilon doesn't quite get there, so use a slightly less epsilon-ey value.
     epsilon = float_info.epsilon * 10
     diff = diff - epsilon
 
     # relies on rounding half away from zero, unlike numpy's bankers' rounding
-    rounded = round(price - (diff if prefer_round_down else -diff), 2)
-    if zp_math.tolerant_equals(rounded, 0.0):
-        return 0.0
+    rounded = round(price - (diff if prefer_round_down else -diff), precision)
+    # if zp_math.tolerant_equals(rounded, 0.0):
+    #     return 0.0
```

Comment:
@prsutherland It could be, but I'm not sure if we may need it - it seems unnecessary now that precision could be as small as desired

Reply:
I believe it is needed to avoid near zero floating point errors. For example, I believe without it, the code will return

```
>>> asymmetric_round_price(.004, True, tick_size=0.005)
-0.0
```

Reply:
Ok, I'll leave it in that case